nodes/vr_sros/vr-sros.go

```diff
@@ -280,19 +280,16 @@ func (s *vrSROS) applyPartialConfig(ctx context.Context, addr, platformName, use
 	// converting byte slice to newline delimited string slice
 	cfgs := strings.Split(string(configContent), "\n")
 
-	mr, err := d.SendConfigs(cfgs)
-	if err != nil || mr.Failed != nil {
-		return fmt.Errorf("failed to apply config; error: %+v %+v", err, mr.Failed)
-	}
 	// condfig snippets should not have commit command, so we need to commit manually
-	r, err := d.SendConfig("commit")
-	if err != nil || r.Failed != nil {
-		return fmt.Errorf("failed to commit config; error: %+v %+v", err, mr.Failed)
-	}
+	cfgs = append(cfgs, "commit", "/admin save")
 
-	r, err = d.SendCommand("/admin save")
-	if err != nil || r.Failed != nil {
-		return fmt.Errorf("failed to persist config; error: %+v %+v", err, mr.Failed)
+	mr, err := d.SendConfigs(cfgs)
+	if err != nil || (mr != nil && mr.Failed != nil) {
+		if mr != nil {
+			return fmt.Errorf("failed to apply config; error: %+v %+v", err, mr.Failed)
```

Comment:
this one is not needed, as we check for `mr != nil` above

Reply:
thats not correct.
its an OR up there ... we could do it in to seperate branches but ... same same

Reply:
then here it shoudl also be `if mr != nil && err != nil`
otherwise the statement above might have err == nil, but mr.Failed != nil